symengine/sympy_compat.py

```diff
@@ -64,158 +64,188 @@ def __new__(cls, name):
         return symengine.UndefFunction(name)
 
 
-class log(Function):
+from types import ModuleType
+
+functions = ModuleType(__name__ + ".functions")
+import sys
+sys.modules[functions.__name__] = functions
+
+functions.sqrt = sqrt
+functions.exp = exp
+
+
+class _FunctionRegistrarMeta(BasicMeta):
+
+    def __new__(mcls, name, bases, dict):
+        cls = BasicMeta.__new__(mcls, name, bases, dict)
+        if not name.startswith("_"):
+            setattr(functions, name, cls)
+        return cls
```

Comment:
Okay, then that's fine. Can you move setting the `name` attribute to `symengine_wrapper.pyx`? Otherwise `(sin(x) + 1) - 1` wouldn't have the name attribute, since it's only added if the function is directly created.